dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -666,14 +666,15 @@ String DAGExpressionAnalyzer::applyFunction(
     const String & func_name,
     const Names & arg_names,
     const ExpressionActionsPtr & actions,
-    const TiDB::TiDBCollatorPtr & collator)
+    const TiDB::TiDBCollatorPtr & collator,
+    const tipb::Expr * expr)
 {
     String result_name = genFuncString(func_name, arg_names, {collator});
     if (actions->getSampleBlock().has(result_name))
         return result_name;
     const FunctionBuilderPtr & function_builder = FunctionFactory::instance().get(func_name, context);
     const ExpressionAction & action = ExpressionAction::applyFunction(function_builder, arg_names, result_name, collator);
```

Comment:
Please check if grouping can use the instance from FunctionFactory directly

Reply:
> Please check if grouping can use the instance from FunctionFactory directly

https://github.com/pingcap/tiflash/blob/aa9174f6c54ebc428e601b2611680817803fbcea/dbms/src/Functions/FunctionFactory.h#L55-L62
According to the above codes, we will create new builder each time we want to get a builder.

Reply:
> Please check if grouping can use the instance from FunctionFactory directly

https://github.com/pingcap/tiflash/blob/master/dbms/src/Functions/FunctionFactory.h#L55-L62
According to the above codes, new function will be created each time we want to get a function.

Reply:
Make sense.